qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -167,6 +177,18 @@ def quantum_instance(self, quantum_instance) -> None:
         elif len(self._circuit.clbits) == 0:
             self._circuit.measure_all()
 
+    @property
+    def input_gradients(self) -> bool:
```

Comment:
Is this purely an internal performance optimization or something that a end user creating the neural network might want to set too. Either way I think it should have a reno to note the performance aspect etc.

Reply:
This is an internal performance optimization. A user may play with it, but at their own risk. A reno file is added.

Reply:
Ok, being public it will show up in API reference hence why I asked.

Reply:
Would `backward_gradients` be a more specific name here rather than input_gradients. The text seems to suggest that it only applies to backwards hence the question

Reply:
`backward_gradients` does not work. There are two types: gradients wrt input and wrt weights. Both are computed in the backward pass. Weights are always computed.